src/xo-app/migrate-vm/index.js

```diff
@@ -0,0 +1,245 @@
+
+  return (state, props) => ({
+    networks: getNetworks(state, props),
+    pifs: getPifs(state, props),
+    pools: getPools(state, props),
+    srs: getSrs(state, props),
+    vdis: getVdis(state, props),
+    vifs: getVifs(state, props)
+  })
+}, { withRef: true })
+export default class MigrateVmModalBody extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      mapVdisSrs: {},
+      mapVifsNetworks: {}
+    }
+
+    this._getHostPredicate = createSelector(
+      () => this.props.vm,
+      ({ $container }) => host => host.id !== $container
+    )
+
+    this._getSrPredicate = createSelector(
+      () => this.state.host,
+      host => (host
+        ? sr => sr.content_type !== 'iso' && (sr.$container === host.id || sr.$container === host.$pool)
+        : () => false)
+    )
+
+    this._getNetworkPredicate = createSelector(
+      createPicker(
+        () => this.props.pifs,
+        createSelector(
+          () => this.state.host,
+          host => host && host.$PIFs
+        )
+      ),
+      pifs => {
+        if (!pifs) {
+          return false
+        }
+
+        const networks = {}
+        forEach(pifs, pif => {
+          pif.ip && (networks[pif.$network] = true)
+        })
+
+        return network => networks[network.id]
+      }
+    )
+  }
+
+  get value () {
+    return {
+      targetHost: this.state.host && this.state.host.id,
+      mapVdisSrs: this.state.mapVdisSrs,
+      mapVifsNetworks: this.state.mapVifsNetworks,
+      migrationNetwork: this.state.network && this.state.network.id
+    }
+  }
+
+  _selectHost = host => {
+    if (!host) {
+      this.setState({ intraPool: undefined })
+      return
+    }
+    const { networks, pools, pifs, srs, vdis, vifs } = this.props
+    const defaultMigrationNetwork = networks[find(pifs, pif => pif.$host === host.id && pif.management).$network]
+    const defaultSr = srs[pools[host.$pool].default_SR]
+    const defaultNetworks = {}
+    // Default network...
+    forEach(vifs, vif => {
+      // ...is the one which has the same name_label as the VIF's previous network (if it has an IP)...
+      const defaultPif = find(host.$PIFs, pifId => {
+        const pif = pifs[pifId]
+        return pif.ip && networks[vif.$network].name_label === networks[pif.$network].name_label
+      })
+      defaultNetworks[vif.id] = defaultPif && networks[defaultPif.$network]
+      // ...or the first network in the target host networks list that has an IP.
+      if (!defaultNetworks[vif.id]) {
+        defaultNetworks[vif.id] = networks[pifs[find(host.$PIFs, pif => pifs[pif].ip)].$network]
+      }
+    })
+    this.setState({
+      network: defaultMigrationNetwork,
+      defaultNetworks,
+      defaultSr,
+      host,
+      intraPool: this.props.vm.$pool === host.$pool
+    }, () => {
+      if (!this.state.intraPool) {
+        this.refs.network.value = defaultMigrationNetwork
+        forEach(vdis, (_, vdiId) => {
```

Comment:
You might wanna use `vdi.id` instead as it will still works if `vdis` is an array.